File: session-1/src/PeakElement.java
import java.util.Scanner;

public class PeakElement {

    public int findPeakElement(int [] nums){
        int n = nums.length;

        int left = 0, right = n-1;

        while (left < right ){
            int mid = (left + right) / 2;

            if(nums[mid] >= nums[mid+1]){
                right = mid;
            }else{
                left = mid+1;
            }
        }

        return left;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        PeakElement x = new PeakElement();

        int n = scanner.nextInt();

        int[] nums = new int[n];

        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }
        System.out.println(x.findPeakElement(nums));

        scanner.close();
    }
}

File: session-1/src/PeakElement.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        start = 0
        end = len(nums) - 1

        while start <= end:
            mid = start + (end - start) // 2

            if mid < len(nums) - 1 and nums[mid] < nums[mid + 1]:
                start = mid + 1
            elif mid > 0 and nums[mid] < nums[mid - 1]:
                end = mid - 1
            else:
                return mid

        return -1
